--- xxdiff-3.2/src/pixmaps/xxdiff.xpm	2005-10-31 07:17:33.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/xxdiff.xpm	2009-06-17 11:11:54.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * xxdiff_xpm[] = {
+static const char * xxdiff_xpm[] = {
 "190 75 2931 2",
 "  	c None",
 ". 	c #B8CFDD",
--- xxdiff-3.2/src/pixmaps/next_difference.xpm	2005-10-31 07:17:33.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/next_difference.xpm	2009-06-17 11:19:26.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * next_difference_xpm[] = {
+static const char * next_difference_xpm[] = {
 "32 32 2 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/next_unselected_difference.xpm	2005-10-31 07:17:33.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/next_unselected_difference.xpm	2009-06-17 11:20:20.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * next_unselected_difference_xpm[] = {
+static const char * next_unselected_difference_xpm[] = {
 "32 32 2 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/previous_difference.xpm	2009-06-17 11:21:58.000000000 +0100
+++ xxdiff-3.2/src/pixmaps/previous_difference.xpm	2009-06-17 11:21:23.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * previous_difference_xpm[] = {
+static const char * previous_difference_xpm[] = {
 "32 32 2 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/previous_unselected_difference.xpm	2005-10-31 07:17:33.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/previous_unselected_difference.xpm	2009-06-17 11:22:49.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * previous_unselected_difference_xpm[] = {
+static const char * previous_unselected_difference_xpm[] = {
 "32 32 2 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/search.xpm	2005-10-31 07:17:33.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/search.xpm	2009-06-17 11:23:45.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * search_xpm[] = {
+static const char * search_xpm[] = {
 "32 32 2 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/search_backward.xpm	2005-10-31 07:17:37.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/search_backward.xpm	2009-06-17 11:25:06.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * search_backward_xpm[] = {
+static const char * search_backward_xpm[] = {
 "32 32 2 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/search_forward.xpm	2005-10-31 07:17:37.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/search_forward.xpm	2009-06-17 11:25:57.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * search_forward_xpm[] = {
+static const char * search_forward_xpm[] = {
 "32 32 2 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/select_region_left.xpm	2005-10-31 07:17:37.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/select_region_left.xpm	2009-06-17 11:26:53.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * select_region_left_xpm[] = {
+static const char * select_region_left_xpm[] = {
 "32 32 2 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/select_region_middle.xpm	2005-10-31 07:17:37.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/select_region_middle.xpm	2009-06-17 11:28:24.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * select_region_middle_xpm[] = {
+static const char * select_region_middle_xpm[] = {
 "32 32 2 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/select_region_neither.xpm	2005-10-31 07:17:37.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/select_region_neither.xpm	2009-06-17 11:29:34.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * select_region_neither_xpm[] = {
+static const char * select_region_neither_xpm[] = {
 "32 32 2 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/select_region_right.xpm	2005-10-31 07:17:37.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/select_region_right.xpm	2009-06-17 11:27:40.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * select_region_right_xpm[] = {
+static const char * select_region_right_xpm[] = {
 "32 32 2 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/select_region_unselect.xpm	2005-10-31 07:17:37.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/select_region_unselect.xpm	2009-06-17 11:30:32.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * select_region_unselect_xpm[] = {
+static const char * select_region_unselect_xpm[] = {
 "32 32 2 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/split_swap_join.xpm	2005-10-31 07:17:37.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/split_swap_join.xpm	2009-06-17 11:31:36.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * split_swap_join_xpm[] = {
+static const char * split_swap_join_xpm[] = {
 "32 32 2 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/diff_files.xpm	2005-10-31 07:18:48.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/diff_files.xpm	2009-06-17 11:32:31.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * diff_files_xpm[] = {
+static const char * diff_files_xpm[] = {
 "32 32 2 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/diff_files_next.xpm	2005-10-31 07:18:48.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/diff_files_next.xpm	2009-06-17 11:33:30.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * diff_files_next_xpm[] = {
+static const char * diff_files_next_xpm[] = {
 "32 32 2 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/save_as_left.xpm	2005-10-31 07:18:01.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/save_as_left.xpm	2009-06-17 11:34:17.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * save_as_left_xpm[] = {
+static const char * save_as_left_xpm[] = {
 "32 32 52 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/save_as_middle.xpm	2005-10-31 07:18:01.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/save_as_middle.xpm	2009-06-17 11:35:06.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * save_as_middle_xpm[] = {
+static const char * save_as_middle_xpm[] = {
 "32 32 52 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/save_as_right.xpm	2005-10-31 07:18:01.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/save_as_right.xpm	2009-06-17 11:35:48.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * save_as_right_xpm[] = {
+static const char * save_as_right_xpm[] = {
 "32 32 52 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/save_as_merged.xpm	2005-10-31 07:18:01.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/save_as_merged.xpm	2009-06-17 11:37:14.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * save_as_merged_xpm[] = {
+static const char * save_as_merged_xpm[] = {
 "32 32 52 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/save_as.xpm	2005-10-31 07:18:01.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/save_as.xpm	2009-06-17 11:38:08.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * save_as_xpm[] = {
+static const char * save_as_xpm[] = {
 "32 32 52 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/return_accept.xpm	2005-10-31 07:18:48.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/return_accept.xpm	2009-06-17 11:39:08.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * return_accept_xpm[] = {
+static const char * return_accept_xpm[] = {
 "32 32 2 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/return_reject.xpm	2005-10-31 07:18:48.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/return_reject.xpm	2009-06-17 11:39:53.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * return_reject_xpm[] = {
+static const char * return_reject_xpm[] = {
 "32 32 2 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/pixmaps/return_merged.xpm	2005-10-31 07:18:48.000000000 +0000
+++ xxdiff-3.2/src/pixmaps/return_merged.xpm	2009-06-17 11:40:37.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char * return_merged_xpm[] = {
+static const char * return_merged_xpm[] = {
 "32 32 2 1",
 " 	c None",
 ".	c #000000",
--- xxdiff-3.2/src/resParser.cpp	2006-06-11 22:12:33.000000000 +0100
+++ xxdiff-3.2/src/resParser.cpp	2009-06-17 11:51:51.000000000 +0100
@@ -1074,7 +1074,7 @@
       XxCommandSwitch bo = XxCommandSwitch(commandSwitchList[ii]._token);
       int b1 = res1.getInitSwitch( bo );
       XX_CHECK( -1 <= b1 && b1 <= 1 );
-      char* bmap[3] = { "Nop", "False", "True" };
+      const char* bmap[3] = { "Nop", "False", "True" };
       if ( b1 != res2.getInitSwitch( bo ) ) {
          os << initSwitchStr << "." << commandSwitchList[ii]._name << ": "
             << bmap[b1+1] << endl;
@@ -1250,7 +1250,7 @@
       XxCommandSwitch bo = XxCommandSwitch(commandSwitchList[ii]._token);
       int b1 = res.getInitSwitch( bo );
       XX_CHECK( -1 <= b1 && b1 <= 1 );
-      char* bmap[3] = { "Nop", "False", "True" };
+      const char* bmap[3] = { "Nop", "False", "True" };
       os << initSwitchStr << "." << commandSwitchList[ii]._name << ": "
          << bmap[b1+1] << endl;
    }
@@ -1558,7 +1558,7 @@
          XxCommandSwitch bo = XxCommandSwitch(commandSwitchList[ii]._token);
          int b1 = res.getInitSwitch( bo );
          XX_CHECK( -1 <= b1 && b1 <= 1 );
-         char* bmap[3] = { "Nop", "False", "True" };
+         const char* bmap[3] = { "Nop", "False", "True" };
          os << tok->_name << "." << commandSwitchList[ii]._name << ": "
             << bmap[b1+1] << "" << endl;
       }
--- xxdiff-3.2/src/util.cpp	2006-06-11 22:12:33.000000000 +0100
+++ xxdiff-3.2/src/util.cpp	2009-06-17 12:15:48.000000000 +0100
@@ -789,8 +789,8 @@
    }
    if ( pid == 0 ) {
       char* argv[4];
-      argv[0] = "sh";
-      argv[1] = "-c";
+      argv[0] = strdup("sh");
+      argv[1] = strdup("-c");
       argv[2] = const_cast<char*>( command.latin1() );
       argv[3] = 0;
       execve( "/bin/sh", argv, environ );
